### PR TITLE
scenefx: init at 0.1; swayfx-unwrapped: 0.3.2 -> 0.4, decouple from sway-unwrapped and other cleanup

### LINKED ISSUE #307721: Update request: swayfx 0.3.2 -> 0.4
- Package name: swayfx
- Latest released version: 0.4

- Current version on the unstable channel: 0.3.2
- Current version on the stable/release channel: 0.3.2

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@RicArch97 

-----

Note for maintainers: Please tag this issue in your PR.

---
[release note](https://github.com/WillPower3309/swayfx/releases/tag/0.4)

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

Fixes #307721.

- **sway: remove `with lib;`**
- **sway-unwrapped: remove unused arg**
- **sway-unwrapped: use meson string functions**
- **sway-unwrapped: remove `meta = with lib;`**
- **swayfx-unwrapped: remove `meta = with lib;`**
- **swayfx-unwrapped: use `finalAttrs` instead of `rec`**
- **swayfx-unwrapped: fix `meta.changelog`**
- **swayfx-unwrapped: decouple from sway-unwrapped**
- **scenefx: init at 0.1**
- **swayfx-unwrapped: 0.3.2 -> 0.4**
- **swayfx-unwrapped: add version test**
- **swayfx-unwrapped: add basic nixos test**

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by RicArch97 — APPROVED** [review]
Nice work! Looks good to me

Gave it a good test (in addition to the testing already done before the release) using your branch in my personal config. Everything builds and works as expected.

**Comment by eclairevoyant** [comment]
@ofborg test sway swayfx
@ofborg build swayfx-unwrapped.passthru.tests.version scenefx.passthru.tests.pkg-config

File: pkgs/by-name/sw/sway/package.nix [hunk 1]
```diff
@@ -13,9 +13,12 @@
 
 assert extraSessionCommands != "" -> withBaseWrapper;
 
-with lib;
-
 let
+  inherit (builtins) replaceStrings;
```
Comment (Synthetica9) [inline pkgs/by-name/sw/sway/package.nix:17]: Why not from lib?

Comment (eclairevoyant) [reply pkgs/by-name/sw/sway/package.nix:17]: Does it matter?

**Review by ozwaldorf — APPROVED** [review]

**Pull request merged**